### DISCUSSION TITLE
Training efficientdet on image with no object ? (Learning from negative images)

### DISCUSSION THREAD

**Discussion by hieutrluu** [rwightman/efficientdet-pytorch#167] [Q&A] [answered]
I understand that the background category is to 0 by default.
However I am not sure what is the correct way to train on image with no background on custom dataset (Learning from negative images).
Should I set a random number of boxes as ground truth for the data as the entire image is background ?

Because the number of negative images (image with no annotations) is much larger than image with annotations, my pipeline required learning from negative images.
What is the correct way of doing this ?

Best regards

**Comment by rwightman** [comment, marked as answer]
@HieuTrLuu  You shouldn't need to do anything specific for that case, just leave those images with no bbox annotations. The losses should take care of penalizing any predictions for negative images. And the loss config is already designed (retinanet style) to take into account the fact that easy negatives should be downweighted since that's the majority of image space regardless of whether or not you have lots of negative images. 

The only thing you may need to do if you have an abnormally large imbalance, is to tweak tweak some of the loss weightings/params

**Reply by hieutrluu** [reply]
thank you very much for your help and amazing repo

**Comment by dkatsios** [comment]
@rwightman I am not sure I understood your response. What should be the input to the model during training when a batch has both positive and negative images?
